internal/resources/google/sql_database_instance.go

```diff
@@ -0,0 +1,316 @@
+package google
+
+import (
+	"github.com/infracost/infracost/internal/resources"
+	"github.com/infracost/infracost/internal/schema"
+
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/shopspring/decimal"
+)
+
+type SQLDatabaseInstance struct {
+	Address              string
+	DiskSize             int64
+	UseIPV4              bool
+	ReplicaConfiguration string
+	Tier                 string
+	AvailabilityType     string
+	Region               string
+	DatabaseVersion      string
+	DiskType             string
+	BackupStorageGB      *float64 `infracost_usage:"backup_storage_gb"`
+}
+
+var SQLDatabaseInstanceUsageSchema = []*schema.UsageItem{{Key: "backup_storage_gb", ValueType: schema.Float64, DefaultValue: 0}}
+
+func (r *SQLDatabaseInstance) PopulateUsage(u *schema.UsageData) {
+	resources.PopulateArgsWithUsage(r, u)
+}
+
+func (r *SQLDatabaseInstance) BuildResource() *schema.Resource {
+	var resource *schema.Resource
+
+	replica := false
+	if r.ReplicaConfiguration != "" {
+		replica = true
+	}
+
+	resource = sqlDatabaseInstanceCostComponents(r, false, r.Address)
+	if replica {
+		resource.SubResources = append(resource.SubResources, sqlDatabaseInstanceCostComponents(r, true, "Replica"))
+	}
+
+	return resource
+}
+
+type SQLInstanceDBType int
+
+const (
+	MySQL SQLInstanceDBType = iota
+	PostgreSQL
+	SQLServer
+)
+
+func sqlDatabaseInstanceCostComponents(r *SQLDatabaseInstance, replica bool, name string) *schema.Resource {
```

Comment:
I think this can be a method on the struct? Looks like there's nothing else using it.

Reply:
Right, fixed.